src/app/modules/shared/components/details-fields/details-fields.component.ts

```diff
@@ -1,48 +1,76 @@
-import {
-  Component,
-  OnChanges,
-  Input,
-  Output,
-  EventEmitter,
-  ViewChild,
-  ElementRef
-} from '@angular/core';
+import { Component, OnChanges, OnInit, Input, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
 import { FormBuilder, FormGroup } from '@angular/forms';
+import { Store, select } from '@ngrx/store';
+import * as actions from '../../store/technology.actions';
+
+import { allTechnologies } from '../../store/technology.selectors';
+import { Technology } from '../../models';
 
 @Component({
   selector: 'app-details-fields',
   templateUrl: './details-fields.component.html',
-  styleUrls: ['./details-fields.component.scss']
+  styleUrls: ['./details-fields.component.scss'],
 })
-export class DetailsFieldsComponent implements OnChanges {
+export class DetailsFieldsComponent implements OnChanges, OnInit {
   @Input() entryToEdit;
   @Input() formType: string;
   @Output() saveEntry = new EventEmitter();
   @ViewChild('closeModal') closeModal: ElementRef;
   entryForm: FormGroup;
+  technologies: Technology;
```

Comment:
technologies or technology. it looks like technology since the type is not an array or any kind of collection.